lib/iris/fileformats/grib/__init__.py

```diff
@@ -962,7 +970,7 @@ def as_pairs(cube):
 def as_messages(cube):
     """
     Convert one or more cubes to GRIB messages.
-    Returns an iterable of grib_api GRIB messages.
+    Returns an iterable of grib_api GRIB message IDs.
```

Comment:
in my mind, a key deliverable for this change is to ensure that there is no need to use the grib_api directly

So, I would like to put together a proposal which gets all the functionality we need from GribMessage

There is a planned update to as_pairs and as_messages in #1941, which we can adapt to meet this aim

http://scitools.org.uk/iris/docs/v1.9.2/userguide/saving_iris_cubes.html#customising-the-save-process
would look much better if it did not include the gribapi calls, imo